kf/nfconfig.go

```diff
@@ -942,3 +943,311 @@ func getHostInterfaces() (map[string]bool, error) {
+					return fmt.Errorf("failed to chain ingress tc bpf programs: %v", err)
+				}
+				if err := c.PushBackAndStartBPF(bpfProg, ifaceName, models.EgressType); err != nil {
+					return fmt.Errorf("failed to PushBackAndStartBPF BPF Program: %v", err)
+				}
+			}
+		} else if err := c.AddAndStartBPF(bpfProg, ifaceName, models.EgressType); err != nil {
+			return fmt.Errorf("failed to AddAndStartBPF tcegress BPF Program: %v", err)
+		}
+	}
+
+	return nil
+}
+
+// AddeBPFPrograms - Starts eBPF programs on the node if they are not running
+func (c *NFConfigs) AddeBPFPrograms(bpfProgs []models.L3afBPFPrograms) error {
+	for _, bpfProg := range bpfProgs {
+		if err := c.AddProgramsOnInterface(bpfProg.Iface, bpfProg.HostName, bpfProg.BpfPrograms); err != nil {
+			if err := c.SaveConfigsToConfigStore(); err != nil {
+				return fmt.Errorf("add eBPF Programs failed to save configs %v", err)
+			}
+			return fmt.Errorf("failed to Add BPF program on iface %s with error: %v", bpfProg.Iface, err)
+		}
+		c.ifaces = map[string]string{bpfProg.Iface: bpfProg.Iface}
+	}
+	if err := c.SaveConfigsToConfigStore(); err != nil {
+		return fmt.Errorf("AddeBPFPrograms failed to save configs %v", err)
+	}
+	return nil
+}
+
+// DeleteProgramsOnInterface : It will delete ebpf Programs on the given interface
+func (c *NFConfigs) DeleteProgramsOnInterface(ifaceName, HostName string, bpfProgs *models.BPFProgramNames) error {
+	if HostName != c.HostName {
+		errOut := fmt.Errorf("provided bpf programs do not belong to this host")
+		log.Error().Err(errOut)
+		return errOut
+	}
+
+	if ifaceName == "" || bpfProgs == nil {
+		errOut := fmt.Errorf("iface name or bpf programs are empty")
+		log.Error().Err(errOut)
+		return errOut
+	}
+
+	if _, ok := c.hostInterfaces[ifaceName]; !ok {
+		errOut := fmt.Errorf("%s interface name not found in the host", ifaceName)
+		log.Error().Err(errOut)
+		return errOut
+	}
+
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	sort.Strings(bpfProgs.XDPIngress)
+	if c.IngressXDPBpfs[ifaceName] != nil {
+		bpfList := c.IngressXDPBpfs[ifaceName]
+		for e := bpfList.Front(); e != nil; {
+			next := e.Next()
+			data := e.Value.(*BPF)
+			if BinarySearch(bpfProgs.XDPIngress, data.Program.Name) {
+				err := c.DeleteProgramsOnInterfaceHelper(e, ifaceName, models.XDPIngressType, bpfList)
+				if err != nil {
+					return fmt.Errorf("DeleteProgramsOnInterfaceHelper function failed : %v", err)
+				}
+			}
+			e = next
+		}
+	}
+
+	sort.Strings(bpfProgs.TCIngress)
+	if c.IngressTCBpfs[ifaceName] != nil {
+		bpfList := c.IngressTCBpfs[ifaceName]
+		for e := bpfList.Front(); e != nil; {
+			next := e.Next()
+			data := e.Value.(*BPF)
+			if BinarySearch(bpfProgs.TCIngress, data.Program.Name) {
+				err := c.DeleteProgramsOnInterfaceHelper(e, ifaceName, models.IngressType, bpfList)
+				if err != nil {
+					return fmt.Errorf("DeleteProgramsOnInterfaceHelper function failed : %v", err)
+				}
+			}
+			e = next
+		}
+	}
+
+	sort.Strings(bpfProgs.TCEgress)
```

Comment:
Nit: Please break out into a smaller function

Reply:
I think for this it is not possible